lib/kitchen/driver/openstack.rb

```diff
@@ -303,14 +302,22 @@ def get_public_private_ips(server)
       end
 
       def get_ip(server)
-        unless config[:floating_ip].nil?
+        if config[:floating_ip]
           debug "Using floating ip: #{config[:floating_ip]}"
           return config[:floating_ip]
         end
+
+        # make sure we have the latest info
+        info 'Waiting for network information to be available...'
+        w = server.wait_for { reload && !addresses.empty? }
```

Comment:
Resolved this with: https://github.com/test-kitchen/kitchen-openstack/commit/e8023a55cd3dca0ab554428e830f2696ea07a362